packages/core/data-transfer/lib/providers/local-strapi-source-provider/entities.ts

```diff
@@ -58,10 +59,58 @@ export const createEntitiesTransformStream = (): PassThrough => {
 /**
  * Get the list of attributes that needs to be populated for the entities streaming
  */
-const getPopulateAttributes = (contentType: ContentTypeSchema) => {
+const getPopulateAttributes = (strapi: Strapi.Strapi, contentType: ContentTypeSchema) => {
   const { attributes } = contentType;
 
-  return Object.keys(attributes).filter((key) =>
-    ['component', 'dynamiczone'].includes(attributes[key].type)
-  );
+  const populate: any = {};
+
+  const entries: [string, any][] = Object.entries(attributes);
+
+  for (const [key, attribute] of entries) {
+    if (attribute.type === 'component') {
+      const component = strapi.getModel(attribute.component);
+      const subPopulate = getPopulateAttributes(strapi, component);
+
+      if (isArray(subPopulate)) {
+        populate[key] = subPopulate;
+      }
+
+      if (isObject(subPopulate) && size(subPopulate) > 0) {
+        populate[key] = { populate: subPopulate };
+      }
+
+      if (subPopulate === true) {
+        populate[key] = true;
+      }
+    }
+
+    if (attribute.type === 'dynamiczone') {
+      const { components: componentsUID } = attribute;
+
+      const on: any = {};
```

Comment:
Maybe I am lacking some context, but what does this "on" mean, and why is ti useful?

Reply:
Here's more context about the on: https://github.com/strapi/strapi/pull/14879 :slightly_smiling_face: 
Basically it allows you to handle precise populate options for polymorphic data structures (such as a dynamic zone)